### PR TITLE
Add UIKit screenshot tests for layout's `Spacer`

### PR DESCRIPTION
**Pull request by JakeWharton**
This adds the start of a new setup for hosting Kotlin tests within an iOS project so that we can use a Swift screenshot testing library. For now you have to prefix your test functions with "test" and any before/after-annotated functions need to be named "setUp" and "tearDown" (very JUnit 3). We also do not support anything JVM-specific like rules or parameterization for now.

Refs #1343

### CODE REVIEW

File: redwood-layout-uiview/RedwoodLayoutUIViewTests/KotlinHostingXCTestCase.swift [hunk 1]
```diff
@@ -0,0 +1,64 @@
+import Foundation
+import XCTest
+
+/**
+ * An abstract base class for hosting Kotlin test classes in Swift projects.
+ *
+ * Subclasses must override `initTest(name:)` to create instances of `KotlinType`.
+ */
+open class KotlinHostingXCTestCase<KotlinTest>: KotlinHostingXCTestCaseHelper where KotlinTest: NSObject {
```
Comment (JakeWharton) [inline redwood-layout-uiview/RedwoodLayoutUIViewTests/KotlinHostingXCTestCase.swift:9]: Here be crimes

File: redwood-layout-uiview/RedwoodLayoutUIViewTests/UIViewSpacerTestHost.swift [hunk 2]
```diff
@@ -0,0 +1,21 @@
+import SnapshotTesting
+import RedwoodLayoutUIViewTestKt
+import UIKit
+
+final class UIViewSpacerTestHost: KotlinHostingXCTestCase<UIViewSpacerTest> {
+    override class func initTest(name: String) -> UIViewSpacerTest {
+        return UIViewSpacerTest(callback: Callback(named: name))
+    }
+}
```
Comment (JakeWharton) [inline redwood-layout-uiview/RedwoodLayoutUIViewTests/UIViewSpacerTestHost.swift:9]: But this is 👨‍🍳💋

**Review by efirestone — APPROVED** [review]

**Review by colinrtwhite — APPROVED** [review]
Nice!

File: redwood-layout-uiview/RedwoodLayoutUIViewTests/KotlinHostingXCTestCase.swift (hunk 1 above)
Comment (swankjesse) [reply redwood-layout-uiview/RedwoodLayoutUIViewTests/KotlinHostingXCTestCase.swift:9]: 🍿

File: redwood-layout-uiview/RedwoodLayoutUIViewTests/KotlinHostingXCTestCase.swift [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+                // Should be impossible, all types will bottom out in NSObject first.
+                fatalError("For \(KotlinTest.self), type \(targetClass)'s superclass returned nil")
+            }
+            targetClass = nextClass
+        }
+    }
+    
+    private static func createTestMethod(target: KotlinTest, selector: Selector) {
+        let block: @convention(block) () -> Void = {
+            if target.responds(to: #selector(setUp)) {
```
Comment (swankjesse) [inline redwood-layout-uiview/RedwoodLayoutUIViewTests/KotlinHostingXCTestCase.swift:45]: oooooh

File: redwood-layout-uiview/RedwoodLayoutUIViewTests/KotlinHostingXCTestCase.swift [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+                target.perform(#selector(setUp))
+            }
+            defer {
```
Comment (swankjesse) [inline redwood-layout-uiview/RedwoodLayoutUIViewTests/KotlinHostingXCTestCase.swift:48]: neat

**Review by swankjesse — APPROVED** [review]

**Pull request merged**